seahorn/lib/utils.c

```diff
@@ -106,53 +106,68 @@ void assert_byte_cursor_equivalence(
 #define MAX_STRING_LEN MAX_BUFFER_SIZE
 #endif
 
-size_t sea_strlen(const char *str, size_t max) {
-  size_t i;
-  i = nd_size_t();
-  assume(i < max && max <= MAX_STRING_LEN);
-  assume(str[i] == '\0');
+// A strlen implementation
+size_t sea_strlen(const char *str) {
```

Comment:
max must be a parameter otherwise this is not sound implementation. 

alternatively add another loop in case length is not found early. 

return from function instead of break

Reply:
Good point. Fixed.